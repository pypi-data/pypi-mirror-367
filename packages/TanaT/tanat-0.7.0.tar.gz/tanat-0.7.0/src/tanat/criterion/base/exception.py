#!/usr/bin/env python3
"""
Criterion exceptions.
"""

from ...exception import TanatException


class CriterionException(TanatException):
    """
    Base exception for all criterion-related errors in the TanaT framework.
    """


class InvalidCriterionError(CriterionException):
    """
    Raised when an invalid or improperly configured criterion is encountered.
    """
