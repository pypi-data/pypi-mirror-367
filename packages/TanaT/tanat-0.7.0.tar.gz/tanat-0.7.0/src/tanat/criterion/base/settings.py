#!/usr/bin/env python3
"""
Base class for criterion.
"""

import dataclasses

from .enum import CriterionLevel


@dataclasses.dataclass
class Criterion:
    """
    Base class for criterion.
    """

    _REGISTER_NAME = None

    def get_compatibility_levels(self, max_level=CriterionLevel.TRAJECTORY):
        """List the compatibility levels of the criterion.
        Compatibility levels are marked with decorators.

        Args:
            max_level (CriterionLevel, optional): The maximum level to consider.
                Defaults to Trajectory. Entity < Sequence < Trajectory

        Returns a list of CriterionLevel representing the compatibility levels of the
        criterion. The levels are "sequence","entity" or "trajectory".
        """
        if isinstance(max_level, str):
            max_level = CriterionLevel.from_str(max_level)

        compatibility_levels = getattr(self, "COMPATIBILITY_LEVELS", set())
        return [level for level in compatibility_levels if level <= max_level]
