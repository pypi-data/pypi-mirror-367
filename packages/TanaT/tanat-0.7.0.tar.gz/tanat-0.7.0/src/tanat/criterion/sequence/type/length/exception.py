#!/usr/bin/env python3
"""
Length criterion exception.
"""

from ....base.exception import CriterionException


class ContradictoryLengthCriterionError(CriterionException):
    """
    Raised when length-based filtering criterion are logically incompatible.

    This exception occurs during length criterion configuration when:
    - Attempting to use both 'gt' and 'ge' operators simultaneously
    - Attempting to use both 'lt' and 'le' operators simultaneously
    - Specifying lower and upper bounds that create an impossible condition
      (e.g., lower bound is greater than or equal to upper bound)

    Prevents silent failures in sequence length filtering by explicitly
    highlighting configuration errors.
    """
