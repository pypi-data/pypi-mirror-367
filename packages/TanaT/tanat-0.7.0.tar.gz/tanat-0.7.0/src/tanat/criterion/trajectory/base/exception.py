#!/usr/bin/env python3
"""
Trajectory Criterion exceptions.
"""

from pypassist.mixin.registrable.exceptions import UnregisteredTypeError

from ...base.exception import CriterionException


class TrajectoryCriterionException(CriterionException):
    """
    Trajectory criterion exceptions.
    """


class UnregisteredTrajectoryCriterionTypeError(
    UnregisteredTypeError, TrajectoryCriterionException
):
    """
    Unregistered trajectory criterion type error.
    """
