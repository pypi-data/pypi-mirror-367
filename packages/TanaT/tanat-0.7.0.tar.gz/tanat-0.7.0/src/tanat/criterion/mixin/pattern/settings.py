#!/usr/bin/env python3
"""
Pattern criterion.
"""

from typing import Union

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer
from pypassist.fallback.typing import Dict, List

from ...base.settings import Criterion
from ...decorator.entity import entity_compatible
from ...decorator.sequence import sequence_compatible
from .enum import PatternOperator


@viewer
@entity_compatible
@sequence_compatible
@dataclass
class PatternCriterion(Criterion):
    """
    Flexible pattern-based filtering for sequences and entities.

    Enables complex pattern matching across data columns with advanced
    configuration options.

    Args:
        pattern (Dict[str, Union[str, List[str]]]):
            Column patterns to match. Supports:
            - Plain strings
            - Lists of strings
            - Regex patterns (prefix with 'regex:')

        contains (bool, optional):
            - True: Include sequences matching pattern
            - False: Exclude sequences matching pattern
            Defaults to True.

        case_sensitive (bool, optional):
            Whether matching is case-sensitive. Defaults to True.

        operator (PatternOperator, optional):
            Logical operator between patterns:
            - AND: All patterns must match
            - OR: At least one pattern matches
            Defaults to AND.

    Example:
        # Find sequences with emergency events
        PatternCriterion(pattern={'event_type': 'EMERGENCY'})
    """

    pattern: Dict[str, Union[str, List[str]]]
    contains: bool = True
    case_sensitive: bool = True
    operator: PatternOperator = PatternOperator.AND

    _REGISTER_NAME = "pattern"
