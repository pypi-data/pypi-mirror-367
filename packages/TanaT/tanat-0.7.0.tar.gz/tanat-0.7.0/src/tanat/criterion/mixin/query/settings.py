#!/usr/bin/env python3
"""
Settings for query criterion.
"""

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer

from ...base.settings import Criterion
from ...decorator.entity import entity_compatible
from ...decorator.sequence import sequence_compatible


@viewer
@entity_compatible
@sequence_compatible
@dataclass
class QueryCriterion(Criterion):
    """
    Pandas-based filtering criterion for sequences and entities.

    Allows complex data filtering using pandas query syntax. Compatible with
    sequence and entity levels, enabling flexible data selection based on
    column conditions.

    Args:
        query (str): Pandas-style query string to filter data.
            Supports complex boolean conditions across columns.

    Example:
        # Filter sequences where age > 50 and gender is 'M'
        criterion = QueryCriterion(query="age > 50 and gender == 'M'")
    """

    query: str

    _REGISTER_NAME = "query"
