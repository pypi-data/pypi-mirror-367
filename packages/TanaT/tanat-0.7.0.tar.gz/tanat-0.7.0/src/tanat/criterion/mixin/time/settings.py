#!/usr/bin/env python3
"""
Settings for time-based filtering.
"""

from typing import Optional, Union
from datetime import datetime

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer

from ...base.settings import Criterion
from ...decorator.entity import entity_compatible
from ...decorator.sequence import sequence_compatible


@viewer
@entity_compatible
@sequence_compatible
@dataclass
class TimeCriterion(Criterion):
    """
    Flexible time-based filtering for sequences and entities.

    Enables precise temporal filtering with multiple time constraints and
    different containment modes depending on the filtering level.

    Args:
        start_before (datetime, optional):
            Filter sequences or entities starting before this timestamp.
        start_after (datetime, optional):
            Filter sequences or entities starting after this timestamp.
        end_before (datetime, optional):
            Filter sequences or entities ending before this timestamp.
        end_after (datetime, optional):
            Filter sequences or entities ending after this timestamp.
        duration_within (bool, optional):
            For entity-level filtering on state/interval entities:
            - True: Entity must be entirely contained within the time range
            - False: Entity can partially overlap with the time range
            Defaults to False. Has no effect on event entities.
        sequence_within (bool, optional):
            For sequence-level filtering (event, interval, state):
            - True: Entire sequence must be contained within the time range
            - False: Sequence can partially overlap with the time range
            Defaults to False.
        date_format (str, optional):
            Custom datetime parsing format for string inputs.

    Examples:
        # Filter entities with partial overlap (default behavior)
        TimeCriterion(
            start_after=datetime.now(),
            end_before=datetime.now() + timedelta(days=90)
        )

        # Only keep state/interval entities fully contained in time range
        TimeCriterion(
            start_after=datetime(2024, 1, 1),
            end_before=datetime(2024, 12, 31),
            duration_within=True
        )

        # Only keep sequences entirely within time range
        TimeCriterion(
            start_after=datetime(2024, 6, 1),
            end_before=datetime(2024, 6, 30),
            sequence_within=True
        )

        # Combine both strict modes
        TimeCriterion(
            start_after=datetime(2024, 1, 1),
            end_before=datetime(2024, 12, 31),
            duration_within=True,
            sequence_within=True
        )
    """

    start_before: Optional[Union[datetime, str, int]] = None
    start_after: Optional[Union[datetime, str, int]] = None
    end_before: Optional[Union[datetime, str, int]] = None
    end_after: Optional[Union[datetime, str, int]] = None
    duration_within: bool = False
    sequence_within: bool = False
    date_format: Optional[str] = None
    # TODO: add settings for duration

    _REGISTER_NAME = "time"
