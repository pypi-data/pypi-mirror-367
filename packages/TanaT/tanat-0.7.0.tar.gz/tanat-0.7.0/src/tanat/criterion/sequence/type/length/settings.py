#!/usr/bin/env python3
"""
Length criterion.
"""

from typing import Optional

from pydantic.dataclasses import dataclass
from pypassist.dataclass.decorators.viewer.decorator import viewer

from ....base.settings import Criterion
from ....decorator.sequence import sequence_compatible
from .exception import ContradictoryLengthCriterionError


@viewer
@sequence_compatible
@dataclass
class LengthCriterion(Criterion):
    """
    Filter sequences based on their length.

    Provides flexible length-based filtering with multiple comparison
    operators for sequence selection.

    Args:
        gt (int, optional):
            Keep sequences with length strictly greater than this value.
        ge (int, optional):
            Keep sequences with length greater than or equal to this value.
        lt (int, optional):
            Keep sequences with length strictly less than this value.
        le (int, optional):
            Keep sequences with length less than or equal to this value.

    Example:
        # Filter sequences with 10 or fewer events
        LengthCriterion(le=10)
    """

    gt: Optional[int] = None  # greater than
    ge: Optional[int] = None  # greater than or equal to
    lt: Optional[int] = None  # less than
    le: Optional[int] = None  # less than or equal to

    _REGISTER_NAME = "length"

    def __post_init__(self):
        if self.gt is not None and self.ge is not None:
            raise ContradictoryLengthCriterionError(
                "Cannot use 'gt' and 'ge' operators simultaneously"
            )
        if self.lt is not None and self.le is not None:
            raise ContradictoryLengthCriterionError(
                "Cannot use 'lt' and 'le' operators simultaneously"
            )

        lower = self.gt if self.gt is not None else self.ge
        upper = self.lt if self.lt is not None else self.le
        if lower is not None and upper is not None and lower >= upper:
            raise ContradictoryLengthCriterionError(
                f"lower bound ({lower}) must be strictly less than upper bound ({upper})."
            )
