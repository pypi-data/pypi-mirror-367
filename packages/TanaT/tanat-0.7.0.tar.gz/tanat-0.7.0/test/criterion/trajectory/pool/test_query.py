#!/usr/bin/env python3
"""
Test query criterion applied to trajectory pools.
"""

import pytest

from tanat.criterion.mixin.query.settings import QueryCriterion
from tanat.criterion.base.exception import InvalidCriterionError


class TestQueryCriterion:
    """
    Test query criterion applied to trajectory pools.
    """

    @pytest.mark.parametrize(
        "sequence_name,query",
        [
            ("event", "event_type == 'EMERGENCY'"),
            ("state", "health_state == 'TREATMENT'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_filter_trajectories_on_sequence_level(
        self, trajectory_pool, sequence_name, query, snapshot
    ):
        """
        Trajectory pool: Query criterion applied to sequence level (inplace=False).
        """
        query_criterion = QueryCriterion(query=query)
        filtered_pool = trajectory_pool.filter(
            criterion=query_criterion,
            level="sequence",
            sequence_name=sequence_name,
            inplace=False,
            intersection=True,
        )
        snapshot.assert_match(sorted(filtered_pool.unique_ids))
        assert filtered_pool.settings.intersection is True

    @pytest.mark.parametrize(
        "sequence_name,query",
        [
            ("event", "event_type == 'EMERGENCY'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_filter_trajectories_on_entity_level(
        self, trajectory_pool, sequence_name, query, snapshot
    ):
        """
        Test filtering trajectories on entity level.
        """
        query_criterion = QueryCriterion(query=query)
        filtered_pool = trajectory_pool.filter(
            criterion=query_criterion,
            level="entity",
            sequence_name=sequence_name,
            inplace=False,
            intersection=True,
        )

        snapshot.assert_match(sorted(filtered_pool.unique_ids))
        assert filtered_pool.settings.intersection is True

    def test_filter_entity_level_raises_error_on_state(self, trajectory_pool):
        """
        Test filtering entity level raises error on state.
        """
        criterion = QueryCriterion(query="health_state == 'TREATMENT'")
        with pytest.raises(NotImplementedError):
            trajectory_pool.filter(
                criterion=criterion,
                level="entity",
                sequence_name="state",
                inplace=False,
                intersection=True,
            )

    @pytest.mark.parametrize(
        "sequence_name,query",
        [
            ("event", "event_type == 'EMERGENCY'"),
            ("state", "health_state == 'TREATMENT'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_filter_trajectories_on_sequence_level_inplace(
        self, trajectory_pool, sequence_name, query, snapshot
    ):
        """
        Trajectory pool: Query criterion applied to sequence level (inplace=True).
        """
        query_criterion = QueryCriterion(query=query)

        ## -- filtering on copy
        trajectory_pool_copy = trajectory_pool.copy()
        filtered_pool = trajectory_pool_copy.filter(
            criterion=query_criterion,
            level="sequence",
            sequence_name=sequence_name,
            inplace=True,
            intersection=True,
        )

        ## -- inplace should return None
        assert filtered_pool is None
        snapshot.assert_match(sorted(trajectory_pool_copy.unique_ids))

    @pytest.mark.parametrize(
        "sequence_name,query",
        [
            ("event", "event_type == 'EMERGENCY'"),
            ("state", "health_state == 'TREATMENT'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_filter_trajectories_on_sequence_level_consistency(
        self, trajectory_pool, sequence_name, query
    ):
        """
        Check filter sequence level with inplace=True and inplace=False produce the same results.
        """
        ## -- query criterion
        query_criterion = QueryCriterion(query=query)

        # Non-inplace filtering
        filtered_pool = trajectory_pool.filter(
            criterion=query_criterion,
            level="sequence",
            sequence_name=sequence_name,
            inplace=False,
            intersection=True,
        )
        non_inplace_result = sorted(filtered_pool.unique_ids)

        # Inplace filtering
        traj_pool_copy = trajectory_pool.copy()
        traj_pool_copy.filter(
            criterion=query_criterion,
            level="sequence",
            sequence_name=sequence_name,
            inplace=True,
            intersection=True,
        )
        inplace_result = sorted(traj_pool_copy.unique_ids)

        # Test consistency
        assert non_inplace_result == inplace_result

    @pytest.mark.parametrize(
        "sequence_name,query",
        [
            ("event", "event_type.isin(['EMERGENCY', 'SPECIALIST'])"),
            ("state", "health_state.isin(['TREATMENT', 'CONVALESCENCE'])"),
            ("interval", "medication.isin(['ANTIBIOTIC', 'PAIN_RELIEVER'])"),
        ],
    )
    def test_filter_trajectories_with_multiple_conditions(
        self, trajectory_pool, sequence_name, query, snapshot
    ):
        """
        Test filtering with multiple conditions in the query.
        """
        ## -- query criterion with multiple conditions
        query_criterion = QueryCriterion(query=query)

        # Non-inplace filtering
        filtered_pool = trajectory_pool.filter(
            criterion=query_criterion,
            level="sequence",
            sequence_name=sequence_name,
            inplace=False,
            intersection=True,
        )
        snapshot.assert_match(sorted(filtered_pool.unique_ids))

    @pytest.mark.parametrize(
        "sequence_name,query",
        [
            ("event", "event_type == 'NONEXISTENT_EVENT'"),
            ("state", "health_state == 'NONEXISTENT_STATE'"),
            ("interval", "medication == 'NONEXISTENT_MEDICATION'"),
        ],
    )
    def test_filter_no_matching_query_criterion(
        self, trajectory_pool, sequence_name, query
    ):
        """
        Test filtering with query criterion that doesn't match any data.
        Should return an empty trajectory pool.
        """
        ## -- query criterion with non-existent value
        query_criterion = QueryCriterion(query=query)

        # Non-inplace filtering
        filtered_pool = trajectory_pool.filter(
            criterion=query_criterion,
            level="sequence",
            sequence_name=sequence_name,
            inplace=False,
            intersection=True,
        )

        assert filtered_pool.unique_ids == set()

    @pytest.mark.parametrize(
        "sequence_name,query",
        [
            ("event", "event_type == 'NONEXISTENT_EVENT'"),
            ("state", "health_state == 'NONEXISTENT_STATE'"),
            ("interval", "medication == 'NONEXISTENT_MEDICATION'"),
        ],
    )
    def test_filter_no_matching_query_criterion_inplace(
        self, trajectory_pool, sequence_name, query
    ):
        """
        Test filtering with query criterion that doesn't match any data (inplace=True).
        Should result in an empty trajectory pool.
        """
        ## -- query criterion with non-existent value
        query_criterion = QueryCriterion(query=query)

        # Inplace filtering
        traj_pool_copy = trajectory_pool.copy()
        traj_pool_copy.filter(
            criterion=query_criterion,
            level="sequence",
            sequence_name=sequence_name,
            inplace=True,
            intersection=True,
        )

        # Test empty result
        assert traj_pool_copy.unique_ids == set()

    @pytest.mark.parametrize(
        "sequence_name,query",
        [
            ("event", "event_type == 'EMERGENCY'"),
            ("state", "health_state == 'TREATMENT'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_query_filtering_via_dict_criterion(
        self, trajectory_pool, sequence_name, query
    ):
        """
        Verify consistency of filtering using criterion defined in a dictionary.
        """
        # Dictionary criterion
        dict_criterion = {"query": query}

        # Non-inplace filtering with dict criterion
        filtered_pool = trajectory_pool.filter(
            criterion=dict_criterion,
            criterion_type="query",
            level="sequence",
            sequence_name=sequence_name,
            inplace=False,
            intersection=True,
        )
        non_inplace_result = sorted(filtered_pool.unique_ids)

        # Inplace filtering with dict criterion
        traj_pool_copy = trajectory_pool.copy()
        traj_pool_copy.filter(
            criterion=dict_criterion,
            criterion_type="query",
            level="sequence",
            sequence_name=sequence_name,
            inplace=True,
            intersection=True,
        )
        inplace_result = sorted(traj_pool_copy.unique_ids)

        # Test consistency
        assert non_inplace_result == inplace_result

    @pytest.mark.parametrize(
        "sequence_name,query",
        [
            ("event", "event_type == 'EMERGENCY'"),
            ("state", "health_state == 'TREATMENT'"),
            ("interval", "medication == 'ANTIBIOTIC'"),
        ],
    )
    def test_intersection_false(self, trajectory_pool, sequence_name, query, snapshot):
        """
        Test that setting intersection=False does not affect the length of the filtered
        trajectory pool, but does affect the length of the inner sequence pool.
        """
        query_criterion = QueryCriterion(query=query)
        filtered_pool = trajectory_pool.filter(
            criterion=query_criterion,
            level="sequence",
            sequence_name=sequence_name,
            inplace=False,
            intersection=False,
        )

        assert len(filtered_pool) == len(trajectory_pool)
        assert filtered_pool.settings.intersection is False
        filtered_sequence_ids = sorted(
            filtered_pool.sequence_pools[sequence_name].unique_ids
        )
        snapshot.assert_match(filtered_sequence_ids)

    def test_which_method_raises_error(self, trajectory_pool):
        """
        Which method should raise an error for query criterion because this criterion is not
        applicable at 'trajectory' level.
        """

        criterion_kwargs = {"query": 'FAKE_COLUMN == "dummy_value"'}
        criterion = QueryCriterion(**criterion_kwargs)

        with pytest.raises(InvalidCriterionError):
            trajectory_pool.which(criterion)

        ## -- from dict
        with pytest.raises(InvalidCriterionError):
            trajectory_pool.which(criterion_kwargs, criterion_type="query")
