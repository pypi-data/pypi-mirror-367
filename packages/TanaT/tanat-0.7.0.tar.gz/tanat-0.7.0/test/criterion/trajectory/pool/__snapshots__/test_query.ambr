# serializer version: 1
# name: TestQueryCriterion.test_filter_trajectories_on_entity_level[event-event_type == 'EMERGENCY']
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    10,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_on_entity_level[interval-medication == 'ANTIBIOTIC']
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_on_sequence_level[event-event_type == 'EMERGENCY']
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    10,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_on_sequence_level[interval-medication == 'ANTIBIOTIC']
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_on_sequence_level[state-health_state == 'TREATMENT']
  list([
    6,
    7,
    8,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_on_sequence_level_inplace[event-event_type == 'EMERGENCY']
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    10,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_on_sequence_level_inplace[interval-medication == 'ANTIBIOTIC']
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_on_sequence_level_inplace[state-health_state == 'TREATMENT']
  list([
    6,
    7,
    8,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_with_multiple_conditions[event-event_type.isin(['EMERGENCY', 'SPECIALIST'])]
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    10,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_with_multiple_conditions[interval-medication.isin(['ANTIBIOTIC', 'PAIN_RELIEVER'])]
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
  ])
# ---
# name: TestQueryCriterion.test_filter_trajectories_with_multiple_conditions[state-health_state.isin(['TREATMENT', 'CONVALESCENCE'])]
  list([
    3,
    4,
    5,
    6,
    7,
    8,
    9,
  ])
# ---
# name: TestQueryCriterion.test_intersection_false[event-event_type == 'EMERGENCY']
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    10,
  ])
# ---
# name: TestQueryCriterion.test_intersection_false[interval-medication == 'ANTIBIOTIC']
  list([
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
  ])
# ---
# name: TestQueryCriterion.test_intersection_false[state-health_state == 'TREATMENT']
  list([
    6,
    7,
    8,
  ])
# ---
